src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/OrientationBiasFilterer.java

```diff
@@ -294,11 +297,12 @@ static int calculateTotalNumToCut(final double fdrThreshold, final long unfilter
     public static Map<String, SortedMap<Genotype, VariantContext>> createSampleToGenotypeVariantContextSortedMap(final List<String> sampleNames, final Collection<VariantContext> variants) {
 
         // Sorts in reverse order (highest p_artifact goes first and will not allow anything to be equal
-        //  unless they share the same reference)
+        //  unless they share the same reference).  Unfortunately, we cannot check the actual reference, so we rely on
+        //  hashCode and the genotype rendered as a string to be universally unique.
         // Note the negative sign is to sort in reverse error.
         final Comparator<Genotype> genotypePArtifactComparator = Comparator
                 .comparingDouble((Genotype g) -> -OrientationBiasUtils.getGenotypeDouble(g, OrientationBiasFilterConstants.P_ARTIFACT_FIELD_NAME, 0.0))
-                .thenComparingInt(g -> g.hashCode());
+                .thenComparingInt(g -> g.hashCode()).thenComparing(Genotype::toString);
```

Comment:
This still seems like a gross hack to me.  It seems like it would be better to solve this problem properly rather than just kicking it out to less and less probable territory.

Why not create a private subclass of Genotype that includes a static atomic counter that's incremented each time one is created, and then use that in the comparator?